glean-core/src/upload/request.rs

```diff
@@ -28,15 +28,24 @@ impl PingRequest {
     ///
     /// Automatically creates the default request headers.
     /// Clients may add more headers such as `userAgent` to this list.
-    pub fn new(uuid: &str, url: &str, body: JsonValue) -> Self {
+    pub fn new(uuid: &str, path: &str, body: JsonValue) -> Self {
         Self {
             uuid: uuid.into(),
-            url: url.into(),
+            path: path.into(),
             body,
             headers: Self::create_request_headers(),
         }
     }
 
+    pub fn is_deletion_request(&self) -> bool {
+        // The path format should be `/submit/<app_id>/<ping_name>/<schema_version/<doc_id>`
+        self.path
+            .split('/')
+            .nth(3)
+            .expect("URL path does not follow the expected pattern.")
+            == "deletion-request"
```

Comment:
We shouldn't panic on invalid paths, instead treat them as not-a-deletion-ping.
(We don't validate the URL on load, do we?)

```suggestion
            .map(|url| url == "deletion-request)
            .unwrap_or(false)
```

That should work

Reply:
> We don't validate the URL on load, do we?

No.